test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -305,21 +328,40 @@ class ExtAuthzGrpcIntegrationTest : public Grpc::GrpcClientIntegrationParamTest,
           const auto value = std::string(h.value().getStringView());
 
           // This scenario makes sure we have set the headers to be appended later.
-          entry->mutable_append()->set_value(!absl::EndsWith(value, "-first"));
+          if (absl::EndsWith(value, "-first")) {
+            entry->set_append_action(
+                envoy::config::core::v3::HeaderValueOption::OVERWRITE_IF_EXISTS_OR_ADD);
+          } else {
+            entry->set_append_action(
+                envoy::config::core::v3::HeaderValueOption::APPEND_IF_EXISTS_OR_ADD);
+          }
           entry->mutable_header()->set_key(key);
           entry->mutable_header()->set_value(value);
           return Http::HeaderMap::Iterate::Continue;
         });
 
-    for (const auto& response_header_to_add : response_headers_to_append) {
+    for (const auto& response_header_to_add : response_headers_to_add_if_absent) {
       auto* entry = check_response.mutable_ok_response()->mutable_response_headers_to_add()->Add();
       const auto key = std::string(response_header_to_add.first);
       const auto value = std::string(response_header_to_add.second);
 
-      entry->mutable_append()->set_value(true);
+      entry->set_append_action(envoy::config::core::v3::HeaderValueOption::ADD_IF_ABSENT);
+      entry->clear_append();
```

Comment:
IIUC append should not be set here so I think clearing it could theoretically mask issues. Your new code does validate that the append boolean and append_action fields are consistent so lets depend on that to find issues in test.

Reply:
+1 Can remove this and add another test case with none if the "append" or "append_action" set.